sshd-core/src/main/java/org/apache/sshd/common/forward/DefaultForwardingFilter.java

```diff
@@ -979,15 +980,22 @@ public void messageReceived(IoSession session, Readable message) throws Exceptio
                           session, channel, totalMessages, message.available());
             }
 
-            Collection<ClientChannelEvent> result = channel.waitFor(STATIC_IO_MSG_RECEIVED_EVENTS, Long.MAX_VALUE);
-            if (traceEnabled) {
-                log.trace("messageReceived({}) channel={}, count={}, len={} wait result: {}",
-                          session, channel, totalMessages, message.available(), result);
+            OpenFuture future = channel.getOpenFuture();
+            if (future.isOpened()) {
+                OutputStream outputStream = channel.getInvertedIn();
+                outputStream.write(buffer.array(), buffer.rpos(), buffer.available());
+                outputStream.flush();
+            } else {
+                future.addListener(f -> {
+                    try {
+                        OutputStream outputStream = channel.getInvertedIn();
+                        outputStream.write(buffer.array(), buffer.rpos(), buffer.available());
+                        outputStream.flush();
+                    } catch (IOException e) {
+                        channel.getSession().exceptionCaught(e);
+                    }
```

Comment:
I prefer having separate statements for ease of debugging:
```java
} catch (IOException e) {
    Session s = channel.getSession();
    s.exceptionCaught(e);
}
```

Reply:
I am getting suspicious of this line....

Taking my (web browser -> ssh server <-> ssh client -> web server) example, when I kill -9 my **web browser** while it has open connections through the port forward, this line seems to be causing the ssh server to drop the connection to the ssh client....